clients/src/main/java/org/apache/kafka/clients/admin/DescribeReplicaLogDirResult.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.kafka.common.requests.DescribeLogDirsResponse;
+
+
+/**
+ * The result of {@link AdminClient#describeReplicaLogDir(Collection)}.
+ *
+ * The API of this class is evolving, see {@link AdminClient} for details.
+ */
+@InterfaceStability.Evolving
+public class DescribeReplicaLogDirResult {
+    private final Map<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> futures;
+
+    DescribeReplicaLogDirResult(Map<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> futures) {
+        this.futures = futures;
+    }
+
+    /**
+     * Return a map from replica to future which can be used to check the log directory information of individual replicas
+     */
+    public Map<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> values() {
+        return futures;
+    }
+
+    /**
+     * Return a future which succeeds if log directory information of all replicas are available
+     */
+    public KafkaFuture<Map<TopicPartitionReplica, ReplicaLogDirInfo>> all() {
+        return KafkaFuture.allOf(futures.values().toArray(new KafkaFuture[0])).
+            thenApply(new KafkaFuture.Function<Void, Map<TopicPartitionReplica, ReplicaLogDirInfo>>() {
+                @Override
+                public Map<TopicPartitionReplica, ReplicaLogDirInfo> apply(Void v) {
+                    Map<TopicPartitionReplica, ReplicaLogDirInfo> replicaLogDirInfos = new HashMap<>();
+                    for (Map.Entry<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> entry : futures.entrySet()) {
+                        try {
+                            replicaLogDirInfos.put(entry.getKey(), entry.getValue().get());
+                        } catch (InterruptedException | ExecutionException e) {
+                            // This should be unreachable, because allOf ensured that all the futures completed successfully.
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    return replicaLogDirInfos;
+                }
+            });
+    }
+
+    static public class ReplicaLogDirInfo {
+        public final String currentReplicaLogDir;
+        public final String temporaryReplicaLogDir;
+        public final long temporaryReplicaOffsetLag;
+
+        public ReplicaLogDirInfo() {
+            this(null, null, DescribeLogDirsResponse.INVALID_OFFSET_LAG);
+        }
+
+        public ReplicaLogDirInfo(String currentReplicaLogDir, String temporaryReplicaLogDir, long temporaryReplicaOffsetLag) {
+            this.currentReplicaLogDir = currentReplicaLogDir;
+            this.temporaryReplicaLogDir = temporaryReplicaLogDir;
+            this.temporaryReplicaOffsetLag = temporaryReplicaOffsetLag;
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder builder = new StringBuilder();
+            if (temporaryReplicaLogDir != null) {
+                builder.append("ReplicaLogDirInfo(currentReplicaLogDir=")
```

Comment:
I see. I didn't understand the question previously.

I wasn't aware that the conversion is to not print the name fild of the value. Sure. I just changed both the `ReplicaLogDirInfo.toString()` and `ReplicaInfo.toString()` so that they don't print the class name in the string.

BTW, I assume that this conversion only applies to internal classes, which I think is reasonable. For public classes such as `AlterReplicaDirRequest`, the patch will still print the class name in the `toString()` method as we do in e.g. `DeleteTopicsRequest`.

I also added comment to fields in the `ReplicaLogDirInfo` as show below. I think each fields can be interpreted independently and there is no need to explain how to interpret various combinations. For the example you mentioned, if currentLogDir=null, tempLogDir=XXX, offsetsLage=-1, it means the primary replica is not found and there is only a temporary replica for this partition on the given broker, which is possible if the primary replica is in an offline log directory.

```
// The log directory of the primary replica of this partition on the given broker.
// Null if the primary replica is not found for this partition on the given broker.
public final String currentReplicaLogDir;
// The log directory of the temporary replica of this partition on the given broker.
// Null if the temporary replica is not found for this partition on the given broker.
public final String temporaryReplicaLogDir;
// The LEO of the primary replica  - LEO of the temporary replica.
// -1 if either primary or the temporary replica is not found for this partition on the given broker.
public final long temporaryReplicaOffsetLag;
```